### ISSUE TITLE
Crash on start up synchronisation with dmq_usrloc 

### ISSUE THREAD

**Issue by pwarnold** [kamailio/kamailio#244]
Hi,

I've tested 4.3.0, and master as of 99f7d67b20c011e78c277a44ea8cd4d6a820806f and I see the same crash on both.

I have two nodes synchronising with each other. If there are no usrloc entries then everything works fine. The problem occurs on start up when starting one node will cause the other, already running, node to crash. For example, if there are usrloc entries on node A and node B is not running then when node B is started it will cause node A to crash.

Here's the backtrace:

```
#0  0x00007f13d9ac3f50 in core_hash (s1=0x7fff2a04be80, s2=0x0, size=0) at ../../hashes.h:276
276                     v=(*p<<24)+(p[1]<<16)+(p[2]<<8)+p[3];
(gdb) bt
#0  0x00007f13d9ac3f50 in core_hash (s1=0x7fff2a04be80, s2=0x0, size=0) at ../../hashes.h:276
#1  0x00007f13d9ac9a2d in ul_get_aorhash (_aor=0x7fff2a04be80) at urecord.c:805
#2  0x00007f13d9abf83e in unlock_udomain (_d=0x7f13d38d0e08, _aor=0x7fff2a04be80) at udomain.c:1053
#3  0x00007f13d7ea8cad in usrloc_get_all_ucontact (node=0x7f13d38e6860) at usrloc_sync.c:178
#4  0x00007f13d7eac389 in usrloc_dmq_handle_msg (msg=0x7f13d38e9d08, resp=0x7fff2a04c0e0, node=0x7f13d38e6860) at usrloc_sync.c:360
#5  0x00007f13d80d01b9 in worker_loop (id=0) at worker.c:105
#6  0x00007f13d80cdc6a in child_init (rank=0) at dmq.c:293
#7  0x0000000000535958 in init_mod_child (m=0x7f13dbac8350, rank=0) at sr_module.c:898
#8  0x0000000000535682 in init_mod_child (m=0x7f13dbac8778, rank=0) at sr_module.c:895
#9  0x0000000000535c77 in init_child (rank=0) at sr_module.c:924
#10 0x0000000000527733 in main_loop () at main.c:1650
#11 0x000000000052d4cb in main (argc=14, argv=0x7fff2a04c648) at main.c:2533
```

This seems to be a different issue from #236 because the changes in 039231a3b61c00fe36e868417144daa9f04936ba that caused that issue are not in 4.3.0 and also because the fix in 733fd5daf74963b152a2964f9eea1a6fff91858a doesn't help this issue.

Please let me know if I can provide any more information to help with this.

**Comment by charlesrchance** [comment]
Hi,

Please test using latest master at https://github.com/kamailio/kamailio/commit/99f7d67b20c011e78c277a44ea8cd4d6a820806f and if it resolves the issue I will backport to 4.3.

Cheers,
Charles

**Comment by pwarnold** [comment]
Hi Charles,

99f7d67b20c011e78c277a44ea8cd4d6a820806f fixes the garbled text issue (#237) but doesn't fix this problem.

thanks,
Paul

**Comment by charlesrchance** [comment]
Hi Paul,

I've just committed another fix (https://github.com/kamailio/kamailio/commit/bc01d8d6c6abfc7912a27df3ac23bffd818758e9) if you could test and confirm.

Cheers,
Charles

**Comment by pwarnold** [comment]
Hi Charles,

That fixes it. I've successfully tested both master and 4.3.0 patched with 99f7d67 and bc01d8d and both branches now working correctly.

thanks,
Paul

**Comment by charlesrchance** [comment]
Fixed in https://github.com/kamailio/kamailio/commit/bc01d8d6c6abfc7912a27df3ac23bffd818758e9.

Also backported to 4.3.
